File: app/src/main/java/com/example/purepawapp/ui/admin/AdminBookingDetailFragment.java
package com.example.purepawapp.ui.admin;

import android.os.Bundle;
import android.view.View;

import androidx.navigation.fragment.NavHostFragment;

import com.example.purepawapp.data.model.Booking;
import com.example.purepawapp.data.repository.RepoCallback;
import com.example.purepawapp.databinding.FragmentAdminBookingDetailBinding;
import com.example.purepawapp.di.ServiceLocator;
import com.example.purepawapp.ui.common.BaseFragment;
import com.example.purepawapp.util.CurrencyUtils;
import com.example.purepawapp.util.ViewUtils;
import com.google.android.material.chip.Chip;

import java.util.List;

public class AdminBookingDetailFragment extends BaseFragment<FragmentAdminBookingDetailBinding> {

    private String bookingId;

    public AdminBookingDetailFragment() {
        super(FragmentAdminBookingDetailBinding::inflate);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        bookingId = requireArguments().getString("bookingId", "");
        getBinding().btnBack.setOnClickListener(v -> NavHostFragment.findNavController(this).popBackStack());
        loadBooking();
    }

    private void loadBooking() {
        ServiceLocator.getSpaRepository().getAllBookings(new RepoCallback<>() {
            @Override
            public void onSuccess(List<Booking> bookings) {
                Booking booking = null;
                for (Booking b : bookings) {
                    if (b.getId().equals(bookingId)) {
                        booking = b;
                        break;
                    }
                }
                if (booking != null) {
                    bind(booking);
                } else {
                    ViewUtils.toast(AdminBookingDetailFragment.this, "Không tìm thấy lịch hẹn");
                }
            }

            @Override
            public void onError(Exception error) {
                ViewUtils.toast(AdminBookingDetailFragment.this, error.getMessage() != null ? error.getMessage() : "Không thể tải lịch hẹn");
            }
        });
    }

    private void bind(Booking booking) {
        getBinding().tvServiceName.setText(booking.getServiceName());

        StatusStyle style = AdminStatusUi.bookingStatusStyle(booking.getStatus());
        getBinding().tvStatus.setText(style.getLabel());
        getBinding().tvStatus.setTextColor(getResources().getColor(style.getTextColorRes(), null));
        getBinding().tvStatus.setBackgroundTintList(getResources().getColorStateList(style.getBgColorRes(), null));

        getBinding().tvPetName.setText(booking.getPet().getName());
        getBinding().tvPetInfo.setText(booking.getPet().getSpecies() + " · " + booking.getPet().getBreed() + " · " + booking.getPet().getWeightKg() + "kg");
        String note = booking.getPet().getNote();
        getBinding().tvPetNote.setText(note == null || note.isBlank() ? "Không có ghi chú" : note);

        getBinding().tvDatetime.setText(booking.getBookingDate() + " · " + booking.getTimeSlot());
        getBinding().tvPrice.setText(CurrencyUtils.toVndString(booking.getPrice()));

        getBinding().chipGroupStatus.removeAllViews();
        for (String status : AdminStatusUi.BOOKING_STATUS_FLOW) {
            Chip chip = new Chip(requireContext());
            chip.setText(AdminStatusUi.bookingStatusStyle(status).getLabel());
            chip.setCheckable(true);
            chip.setChecked(status.equals(booking.getStatus()));
            chip.setOnClickListener(v -> updateStatus(status));
            getBinding().chipGroupStatus.addView(chip);
        }
    }

    private void updateStatus(String status) {
        ServiceLocator.getSpaRepository().updateBookingStatus(bookingId, status, new RepoCallback<>() {
            @Override
            public void onSuccess(Void result) {
                ViewUtils.toast(AdminBookingDetailFragment.this, "Đã cập nhật trạng thái");
                loadBooking();
            }

            @Override
            public void onError(Exception error) {
                ViewUtils.toast(AdminBookingDetailFragment.this, error.getMessage() != null ? error.getMessage() : "Không thể cập nhật trạng thái");
            }
        });
    }
}

File: app/src/main/java/com/example/purepawapp/ui/admin/AdminBookingDetailFragment.kt
package com.example.purepawapp.ui.admin

import android.os.Bundle
import android.view.View
import androidx.lifecycle.lifecycleScope
import androidx.navigation.fragment.findNavController
import com.example.purepawapp.data.model.Booking
import com.example.purepawapp.databinding.FragmentAdminBookingDetailBinding
import com.example.purepawapp.di.ServiceLocator
import com.example.purepawapp.ui.common.BaseFragment
import com.example.purepawapp.util.toVndString
import com.example.purepawapp.util.toast
import com.google.android.material.chip.Chip
import kotlinx.coroutines.launch

class AdminBookingDetailFragment : BaseFragment<FragmentAdminBookingDetailBinding>(FragmentAdminBookingDetailBinding::inflate) {

    private val bookingId: String by lazy { requireArguments().getString("bookingId").orEmpty() }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.btnBack.setOnClickListener { findNavController().popBackStack() }
        loadBooking()
    }

    private fun loadBooking() {
        viewLifecycleOwner.lifecycleScope.launch {
            ServiceLocator.spaRepository.getAllBookings()
                .onSuccess { bookings ->
                    val booking = bookings.firstOrNull { it.id == bookingId }
                    if (booking != null) bind(booking) else toast("Không tìm thấy lịch hẹn")
                }
                .onFailure { toast(it.message ?: "Không thể tải lịch hẹn") }
        }
    }

    private fun bind(booking: Booking) {
        binding.tvServiceName.text = booking.serviceName

        val style = bookingStatusStyle(booking.status)
        binding.tvStatus.text = style.label
        binding.tvStatus.setTextColor(resources.getColor(style.textColorRes, null))
        binding.tvStatus.backgroundTintList = resources.getColorStateList(style.bgColorRes, null)

        binding.tvPetName.text = booking.pet.name
        binding.tvPetInfo.text = "${booking.pet.species} · ${booking.pet.breed} · ${booking.pet.weightKg}kg"
        binding.tvPetNote.text = booking.pet.note.ifBlank { "Không có ghi chú" }

        binding.tvDatetime.text = "${booking.bookingDate} · ${booking.timeSlot}"
        binding.tvPrice.text = booking.price.toVndString()

        binding.chipGroupStatus.removeAllViews()
        bookingStatusFlow.forEach { status ->
            val chip = Chip(requireContext()).apply {
                text = bookingStatusStyle(status).label
                isCheckable = true
                isChecked = status == booking.status
                setOnClickListener { updateStatus(status) }
            }
            binding.chipGroupStatus.addView(chip)
        }
    }

    private fun updateStatus(status: String) {
        viewLifecycleOwner.lifecycleScope.launch {
            ServiceLocator.spaRepository.updateBookingStatus(bookingId, status)
                .onSuccess {
                    toast("Đã cập nhật trạng thái")
                    loadBooking()
                }
                .onFailure { toast(it.message ?: "Không thể cập nhật trạng thái") }
        }
    }
}
